nan_persistent_12_inl.h

```diff
@@ -93,28 +101,57 @@ class Global : public v8::Global<T> {
 #else
 template<typename T>
 class Global : public v8::UniquePersistent<T> {
+  struct RValue {
+    inline explicit RValue(v8::UniquePersistent<T> *obj) : object_(obj) {}
+    v8::UniquePersistent<T> *object_;
+  };
+
  public:
   inline Global() : v8::UniquePersistent<T>() {}
 
-  template<typename S> inline Global(v8::Local<S> that) :
+  template<typename S> inline explicit Global(v8::Local<S> that) :
     v8::UniquePersistent<T>(v8::Isolate::GetCurrent(), that) {}
 
   template<typename S>
-  inline Global(const v8::PersistentBase<S> &that) :
+  inline explicit Global(const v8::PersistentBase<S> &that) :
       v8::UniquePersistent<S>(v8::Isolate::GetCurrent(), that) {}
 
   inline void Reset() { v8::PersistentBase<T>::Reset(); }
 
-  template <typename S>
+  template<typename S>
   inline void Reset(const v8::Local<S> &other) {
     v8::PersistentBase<T>::Reset(v8::Isolate::GetCurrent(), other);
   }
 
-  template <typename S>
+  template<typename S>
   inline void Reset(const v8::PersistentBase<S> &other) {
     v8::PersistentBase<T>::Reset(v8::Isolate::GetCurrent(), other);
   }
 
+#if NODE_MODULE_VERSION == NODE_0_12_MODULE_VERSION
+  inline void Empty() {
+    reinterpret_cast<v8::Persistent<T>*>(this)->ClearAndLeak();
```

Comment:
Yes, the only other way would be to directly memset the whole object to 0. So this is the lesser of two evils.

On June 15, 2016 6:19:02 PM GMT+03:00, Ben Noordhuis <notifications@github.com> wrote:
>>    inline void Reset(const v8::PersistentBase<S> &other) {
>>      v8::PersistentBase<T>::Reset(v8::Isolate::GetCurrent(), other);
>>    }
>>  
>> +#if NODE_MODULE_VERSION == NODE_0_12_MODULE_VERSION
>> +  inline void Empty() {
>> +    reinterpret_cast<v8::Persistent<T>*>(this)->ClearAndLeak();
>
>This is bordering on illegal, of course, what with `UniquePersistent`
>not being an instance of `Persistent`.  I assume that's the only way to
>leak the internal pointer?
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/581/files/91b4df4b483ec63571087a40f7005a494b5c857f#r67182496